src/python/pants/task/testrunner_task_mixin.py

```diff
@@ -63,6 +66,106 @@ def execute(self):
       all_targets = self._get_targets()
       self._execute(all_targets)
 
+  def report_test_info(self, scope, target, key, test_info):
+    """
+    Add test information to target information
+    :param string scope: The running scope
+    :param Target target: The target that we want to store the test info under
+    :param list of strings or string key: The key for the info being stored
+    :param dict test_info: The info being stored
+    """
+    if target and scope:
+      address = str(target.address.to_address())
+      target_type = target.type_alias
+      self.context.run_tracker.report_target_info('GLOBAL', address, ['target_type'], target_type)
+      self.context.run_tracker.report_target_info(scope, address, key, test_info)
+
+  @staticmethod
+  def parse_test_info(xml_path, error_handler, additional_testcase_attributes=None):
+    """
+    Parses the junit file for info needed about each test
+    Will include:
+      - test result code
+      - test run time duration
+      - test name
+    :param string xml_path: path of the xml file to be parsed
+    :param function error_handler: error handler function
+    :param list of string additional_testcase_attributes: list of additional testcase attributes to
+           be included in test info.
+    :return: dict of tests
+    """
+    tests_in_path = {}
+    testcase_attributes = additional_testcase_attributes or []
+
+    SUCCESS = 'success'
+    SKIPPED = 'skipped'
+    FAILURE = 'failure'
+    ERROR = 'error'
+
+    _XML_MATCHER = re.compile(r'^TEST-.+\.xml$')
+
+    class ParseError(Exception):
+      """Indicates an error parsing a xml report file."""
+
+      def __init__(self, xml_path, cause):
+        super(ParseError, self).__init__('Error parsing test result file {}: {}'
+          .format(xml_path, cause))
+        self._xml_path = xml_path
+        self._cause = cause
+
+      @property
```

Comment:
I don't think it's necessary to defensively make these `@property`s... if you name the underlying fields `self.xml_path` and `self.cause`, it has the same effect, and you can convert it into an `@property` later if you need to.